nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/runtime/ReloadHelper.java

```diff
@@ -129,14 +132,21 @@ public static void hotReloadPackage(String packageId) {
                   .map(key -> new RollbackOptions(id, key, version.toString()))
                   .map(Rollback::new)
                   .forEachOrdered(installTask.getCommandLog()::add);
-            // write the log
-            File file = pkg.getData().getEntry(LocalPackage.UNINSTALL);
-            installTask.writeLog(file);
         } catch (BundleException | PackageException | ConnectServerError e) {
             throw new NuxeoException("Error while updating studio snapshot", e);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new NuxeoException("Error while downloading studio snapshot", e);
+        } finally {
+            if (pkg != null && installTask != null) {
+                // write the log
+                File file = pkg.getData().getEntry(LocalPackage.UNINSTALL);
+                try {
+                    installTask.writeLog(file);
+                } catch (PackageException e) {
+                    throw new NuxeoException("Error while updating studio snapshot", e);
```

Comment:
Could you rename the message ? here this is not about update.
Also, if something wrong happen during this step, we will loose the first exception (if any).

Reply:
You're right. What do you suggest doing here ? Just ignore the exception ? Or write a log ?

Reply:
I think we need to throw it, but without loosing the possible other one.

Reply:
You shouldn't throw in a `finally` (sonar code smell) as it would hide the main exception if there is one. Just do `log.error(e, e); // don't rethrow inside finally` like we have in other places.